modules/javafx.graphics/src/test/java/test/javafx/css/CssParser_mediaQuery_Test.java

```diff
@@ -0,0 +1,310 @@
+            )),
+            mediaRule.getQueries());
+    }
+
+    @Test
+    void parseMultiConjunction() {
+        Stylesheet stylesheet = new CssParser().parse("""
+            @media (prefers-color-scheme: light) and
+                   (prefers-reduced-motion: reduce) and
+                   (prefers-reduced-transparency: no-preference) {
+                .foo { bar: baz; }
+            }
+            """);
+
+        var mediaRule = RuleHelper.getMediaRule(stylesheet.getRules().getFirst());
+        assertEquals(1, mediaRule.getQueries().size());
+        assertEquals(
+            new ConjunctionExpression(
+                new ConjunctionExpression(
+                    new FunctionExpression<>("prefers-color-scheme", "light", _ -> null, ColorScheme.LIGHT),
+                    new FunctionExpression<>("prefers-reduced-motion", "reduce", _ -> false, true)),
+                new FunctionExpression<>("prefers-reduced-transparency", "no-preference", _ -> false, false)
+            ),
+            mediaRule.getQueries().getFirst()
+        );
+    }
+
+    @Test
+    void parseDisjunction() {
+        Stylesheet stylesheet = new CssParser().parse("""
+            @media (prefers-color-scheme: light) or (prefers-reduced-motion: reduce) {
+                .foo { bar: baz; }
+            }
+            """);
+
+        var mediaRule = RuleHelper.getMediaRule(stylesheet.getRules().getFirst());
+        assertEquals(1, mediaRule.getQueries().size());
+        assertEquals(
+            new DisjunctionExpression(
+                new FunctionExpression<>("prefers-color-scheme", "light", _ -> null, ColorScheme.LIGHT),
+                new FunctionExpression<>("prefers-reduced-motion", "reduce", _ -> null, true)
+            ),
+            mediaRule.getQueries().getFirst());
+    }
+
+    @Test
+    void parseMultiDisjunction() {
+        Stylesheet stylesheet = new CssParser().parse("""
+            @media (prefers-color-scheme: light) or
+                   (prefers-reduced-motion: reduce) or
+                   (prefers-persistent-scrollbars: persistent) {
+                .foo { bar: baz; }
+            }
+            """);
+
+        var mediaRule = RuleHelper.getMediaRule(stylesheet.getRules().getFirst());
+        assertEquals(1, mediaRule.getQueries().size());
+        assertEquals(
+            new DisjunctionExpression(
+                new DisjunctionExpression(
+                    new FunctionExpression<>("prefers-color-scheme", "light", _ -> null, ColorScheme.LIGHT),
+                    new FunctionExpression<>("prefers-reduced-motion", "reduce", _ -> false, true)),
+                new FunctionExpression<>("prefers-persistent-scrollbars", "persistent", _ -> false, true)
+            ),
+            mediaRule.getQueries().getFirst()
+        );
+    }
+
+    @Test
+    void disjunctionAndConjunctionCanNotBeCombined() {
+        Stylesheet stylesheet = new CssParser().parse("""
```

Comment:
could we also test for "or not" and "and not" ?

Reply:
I've added a test.